src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -204,19 +204,24 @@ internal override Symbol GetSpecialTypeMember(SpecialMember member)
             return _compilation.IsMemberMissing(member) ? null : base.GetSpecialTypeMember(member);
         }
 
-        private string GetWellKnownAttributeDataStringField(Func<CommonAssemblyWellKnownAttributeData, string> fieldGetter, string missingValue = null)
+#nullable enable
+        private string? GetWellKnownAttributeDataStringField(Func<CommonAssemblyWellKnownAttributeData, string> fieldGetter, string? missingValue = null, Func<AttributeSyntax, bool>? attributeMatchesOpt = null)
         {
-            string fieldValue = missingValue;
+            string? fieldValue = missingValue;
+
+            var data = attributeMatchesOpt is null
+                ? GetSourceDecodedWellKnownAttributeData()
+                : GetSourceDecodedWellKnownAttributeData(attributeMatchesOpt);
 
-            var data = GetSourceDecodedWellKnownAttributeData();
             if (data != null)
             {
                 fieldValue = fieldGetter(data);
             }
 
-            if ((object)fieldValue == (object)missingValue)
+            if (fieldValue == (object?)missingValue)
             {
-                data = GetNetModuleDecodedWellKnownAttributeData();
+                data = GetNetModuleDecodedWellKnownAttributeData(noCache: attributeMatchesOpt is not null);
```

Comment:
Rikki suggested we don't need to modify any netmodule logic because the three attributes used to compute the public key are assembly-level attributes (they are disallowed on modules).
I'll take a stab at that.

Reply:
> Rikki suggested we don't need to modify any netmodule logic because the three attributes used to compute the public key are assembly-level attributes (they are disallowed on modules).
I'll take a stab at that.

In general, assembly level attributes are allowed in netmodules. Are you saying these specific ones are disallowed? Iven in IL?

Reply:
You're right. I didn't realize that we could have assembly-level attributes in netmodules. Thanks